### ISSUE TITLE
Getter 'minTrailingIndex' was called on null

### ISSUE THREAD

**Issue by jagomf** [fluttercandies/waterfall_flow#18]
I have a `SliverWaterfallFlowDelegateWithMaxCrossAxisExtent` that creates a Waterfall grid of elements.

When the device is rotated from a longer list (portrait) to a shorter list (landscape), the bottom of the list goes empty and you have to scroll up to reach to latest items.

Reproduction:
1.  When the device starts is portrait, the list is rendered as a 1 item list in the cross axis
2. Scroll to the bottom of list.
3. Turn the device to landscape.

Expected:
* The list rebuilds having 2 items in the cross axis and showing the latest 2 or 1 items

Happening:
* The list rebuilds but the latest items are way up high in the list. You have to scroll up to get to the latest items.

Also, this error is thrown:

```
════════ Exception caught by rendering library ═════════════════════════════════
The following NoSuchMethodError was thrown during performLayout():
The getter 'minTrailingIndex' was called on null.
Receiver: null
Tried calling: minTrailingIndex

The relevant error-causing widget was
SliverWaterfallFlow
 lib/…/components/flageventlistscaffold.dart
When the exception was thrown, this was the stack
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      RenderSliverWaterfallFlow.performLayout
 package:waterfall_flow/…/rendering/sliver_waterfall_flow.dart
#2      RenderObject.layout
 package:flutter/…/rendering/object.dart
#3      RenderSliverEdgeInsetsPadding.performLayout
 package:flutter/…/rendering/sliver_padding.dart
#4      RenderSliverPadding.performLayout
 package:flutter/…/rendering/sliver_padding.dart
...
The following RenderObject was being processed when the exception was fired: RenderSliverWaterfallFlow#aa835 relayoutBoundary=up3 NEEDS-LAYOUT NEEDS-COMPOSITING-BITS-UPDATE
RenderObject: RenderSliverWaterfallFlow#aa835 relayoutBoundary=up3 NEEDS-LAYOUT NEEDS-COMPOSITING-BITS-UPDATE
    needs compositing
    parentData: paintOffset=Offset(4.0, 0.0) (can use size)
    constraints: SliverConstraints(AxisDirection.down, GrowthDirection.forward, ScrollDirection.idle, scrollOffset: 6661.4, remainingPaintExtent: 334.0, crossAxisExtent: 742.0, crossAxisDirection: AxisDirection.right, viewportMainAxisExtent: 334.0, remainingCacheExtent: 834.0, cacheOrigin: -246.0)
    geometry: SliverGeometry(scrollExtent: 7229.4, paintExtent: 568.0, maxPaintExtent: 7229.4, hasVisualOverflow: true, cacheExtent: 814.0)
        scrollExtent: 7229.4
        paintExtent: 568.0
        maxPaintExtent: 7229.4
        hasVisualOverflow: true
        cacheExtent: 814.0
    currently live children: 0 to 17
    child with index 0: RenderIndexedSemantics#f7bcb relayoutBoundary=up4 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        parentData: crossAxisIndex=0;crossAxisOffset=0.0;trailingLayoutOffset=407.9;indexes[0, 2, 4, 6, 8, 10, 12, 14]; index=0; layoutOffset=0.0 (can use size)
        constraints: BoxConstraints(w=368.5, 0.0<=h<=Infinity)
        semantic boundary
        size: Size(368.5, 407.9)
        index: 0
        child: RenderRepaintBoundary#69a4b relayoutBoundary=up5 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
            needs compositing
            parentData: <none> (can use size)
            constraints: BoxConstraints(w=368.5, 0.0<=h<=Infinity)
            size: Size(368.5, 407.9)
            usefulness ratio: no metrics collected yet (never painted)
            child: RenderPadding#8344e relayoutBoundary=up6 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                parentData: <none> (can use size)
                constraints: BoxConstraints(w=368.5, 0.0<=h<=Infinity)
                size: Size(368.5, 407.9)
                padding: EdgeInsets.zero
                textDirection: ltr
                child: RenderSemanticsGestureHandler#2854f relayoutBoundary=up7 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
                    parentData: offset=Offset(0.0, 0.0) (can use size)
                    constraints: BoxConstraints(w=368.5, 0.0<=h<=Infinity)
                    size: Size(368.5, 407.9)
                    gestures: tap
...
════════════════════════════════════════════════════════════════════════════════

```

**Comment by zmtzawqlp** [comment]
please provide version of widget and flutter sdk. a demo is better

**Comment by jagomf** [comment]
waterfall_flow: `^2.0.4`
flutter: 1.22.3

Demo:

Load this in protrait, scroll to bottom, then rotate to landscape.

```dart
import 'package:flutter/material.dart';
import 'package:waterfall_flow/waterfall_flow.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final List<double> _heights = [
    100, 100, 80, 90, 100, 120, 50, 80, 30, 100, 100, 80, 90, 100, 120, 50, 80, 30,
    100, 100, 80, 90, 100, 120, 50, 80, 30, 100, 100, 80, 90, 100, 120, 50, 80, 30,
    100, 100, 80, 90, 100, 120, 50, 80, 30, 100, 100, 80, 90, 100, 120, 50, 80, 30,
  ];
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Waterfall demo'),
      ),
      body: CustomScrollView(
        slivers: [
          SliverSafeArea(
            top: false,
            bottom: false,
            sliver: SliverPadding(
              padding: const EdgeInsets.all(4.0),
              sliver: SliverWaterfallFlow(
                gridDelegate:
                    const SliverWaterfallFlowDelegateWithMaxCrossAxisExtent(
                  maxCrossAxisExtent: 450,
                  mainAxisSpacing: 5,
                  crossAxisSpacing: 5,
                ),
                delegate: SliverChildBuilderDelegate(
                  (BuildContext context, int index) => Card(
                    child: SizedBox(
                      child: Text('Item ${index.toString()}'),
                      height: _heights[index],
                    ),
                  ),
                  childCount: _heights.length,
                ),
              ),
            ),
          )
        ],
      ),
    );
  }
}

```

**Comment by jagomf** [comment]
@zmtzawqlp have you been able to check and research on this?

**Comment by zmtzawqlp** [comment]
sorry  for delay reply, i will fix it at this weekend.

**Comment by jagomf** [comment]
Fixed in 2.0.5
